internal/pkg/cli/app_init.go

```diff
@@ -232,6 +263,22 @@ func (o *initAppOpts) validateAppName(name string) error {
 	if err != nil {
 		var noSuchAppErr *config.ErrNoSuchApplication
 		if errors.As(err, &noSuchAppErr) {
+			roleName := fmt.Sprintf("%s-adminrole", name)
+			tags, err := o.iamRoleManager.ListRoleTags(roleName)
+			// NOTE: This is a best-effort attempt to check if the app exists in other regions.
+			// The error either indicates that the role does not exist, or not.
+			// In the first case, it means that this is a valid app name, hence we don't error out.
+			// In the second case, since this is a best-effort, we don't need to surface the error either.
+			if err != nil {
+				return nil
+			}
+			_, hasTag := tags[deploy.AppTagKey]
+			if hasTag {
+				return &errAppAlreadyExistsInAccount{appName: name}
+			}
+			if !hasTag {
+				return &errStackSetAdminRoleExistsInAccount{roleName: roleName}
+			}
```

Comment:
I think we can get rid of the `if !hasTag` since it'd always be true when it reaches here
```suggestion
			return &errStackSetAdminRoleExistsInAccount{roleName: roleName}
		
```